### PR TITLE
Catch SIGTERM and safe launchd stop

### PR DESCRIPTION
**Pull request by gabriel**
`launchctl unload` uses SIGKILL (for fast termination)... and changing to use `launchctl remove` which uses SIGTERM.

Also fixed use of logger and using terminal UI for launchd command output.

Also see https://keybase.atlassian.net/browse/DESKTOP-285

### CODE REVIEW

**Comment by gabriel** [comment]
cc @cjb @patrickxb

File: go/keybase/main.go [hunk 1]
```diff
@@ -239,7 +240,7 @@ func registerGlobalLogUI(g *libkb.GlobalContext) error {
 
 func HandleSignals() {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM, os.Kill)
```
Comment (maxtaco) [inline go/keybase/main.go]: I wonder what this does on Windows?

Comment (patrickxb) [inline go/keybase/main.go]: I'm guessing it should be ok:

ztypes_windows.go:	SIGTERM = Signal(0xf)

File: go/launchd/launchd.go [hunk 2]
```diff
@@ -42,43 +48,71 @@ func (s Service) Label() string { return s.label }
...
+	for i := 0; i < 5; i++ {
+		status, err := s.LoadStatus()
+		if err != nil {
+			return err
+		}
+		if status == nil || !status.IsRunning() {
+			running = false
+			break
+		}
+		if (i+1)%3 == 0 {
```
Comment (patrickxb) [inline go/launchd/launchd.go]: if i == 2?

**Comment by patrickxb** [comment]
This LGTM...

One question: should we add this to our plist files:

     ExitTimeOut <integer>
     The amount of time launchd waits between sending the SIGTERM signal and before sending a SIGKILL signal
     when the job is to be stopped. The default value is system-defined. The value zero is interpreted as
     infinity and should not be used, as it can stall system shutdown forever.

No clue what the default value is, but it looks like 5 seconds is the time that this code waits for it to exit, so maybe we should make sure ExitTimeOut is around 5 seconds as well?

File: go/launchd/launchd.go (hunk 2 above)
Comment (gabriel) [inline go/launchd/launchd.go]: Yeah was trying to not log when i == 0.... changed this in the next commit though so it's moot

**Comment by gabriel** [comment]
I changed the wait timeout to use a duration and set a large timeout (25 seconds) which is > default which docs says is 20...

In practice it looks like launchd waits ~5 seconds after SIGTERM before doing SIGKILL so basically this only applies for zombied process.

Put all this in a comment above the wait timeout.

**Comment by cjb** [comment]
@gabriel :thumbsup: shall we go ahead and land this now?

**Comment by gabriel** [comment]
yep was waiting for travis

**Pull request merged**